connexion/operations/abstract.py

```diff
@@ -199,7 +205,12 @@ def _query_args_helper(self, query_defns, query_arguments,
                     logger.error("Function argument '{}' not defined in specification".format(key))
                 else:
                     logger.debug('%s is a %s', key, query_defn)
-                    res[key] = self._get_val_from_param(value, query_defn)
+                    if isinstance(value, dict):
+                        if key not in res:
+                            res[key] = {}
+                        res[key].update(self._get_val_from_object_param(value, query_defn))
```

Comment:
if this is the only place where this is called, I would not add another function here, and instead let _get_val_from_param handle the type checking and returning the correct response.